src/main/java/com/mixpanel/android/mpmetrics/MixpanelPushNotification.java

```diff
@@ -290,31 +289,193 @@ protected void maybeAddActionButtons() {
-        return (new Notification.Action.Builder(icon, title, createActionIntent(uri))).build();
+    protected List<NotificationButtonData> buildButtons(String buttonsJsonStr) {
+        List<NotificationButtonData> buttons = new ArrayList<>();
+        if (null != buttonsJsonStr) {
+            try {
+                JSONArray buttonsArr = new JSONArray(buttonsJsonStr);
+                for (int i = 0; i < buttonsArr.length(); i++) {
+                    JSONObject buttonObj = buttonsArr.getJSONObject(i);
+
+                    // get button icon from name if one sent
+                    int btnIcon = NotificationData.NOT_SET;
+                    if (buttonObj.has("icnm")) {
+                        String btnIconName = buttonObj.getString("icnm");
+                        if (drawableIds.knownIdName(btnIconName)) {
+                            btnIcon = drawableIds.idFromName(btnIconName);
+                        }
+                    }
+
+                    // handle button label
+                    final String btnLabel = buttonObj.getString("lbl");
+
+                    // handle button action
+                    final JSONObject pushActionJSON = buttonObj.getJSONObject("ontap");
+                    final PushTapTarget target = PushTapTarget.fromString(pushActionJSON.getString("type"));
+                    final PushTapAction pushAction = new PushTapAction(target, pushActionJSON.getString("uri"));
+
+                    //handle button id
+                    final String btnId = buttonObj.getString("id");
+
+                    buttons.add(new NotificationButtonData(btnIcon, btnLabel, pushAction, btnId));
+                }
+            } catch (JSONException e) {
+                MPLog.e(LOGTAG, "Exception parsing buttons payload", e);
+            }
+        }
+
+        return buttons;
     }
 
-    protected PendingIntent createActionIntent(String uri) {
-        return PendingIntent.getActivity(
-                context,
-                0,
-                new Intent(Intent.ACTION_VIEW, Uri.parse(uri)),
-                PendingIntent.FLAG_UPDATE_CURRENT
-        );
+    protected PushTapAction buildOnTap(String onTapStr) {
+        PushTapAction onTap = null;
+        if (null != onTapStr) {
+            try {
+                final JSONObject onTapJSON = new JSONObject(onTapStr);
+                final String typeFromJSON = onTapJSON.getString("type");
+
+                if (!typeFromJSON.equals(PushTapTarget.HOMESCREEN.getTarget())) {
+                    final String uriFromJSON = onTapJSON.getString("uri");
+                    onTap = new PushTapAction(PushTapTarget.fromString(typeFromJSON), uriFromJSON);
+                } else {
+                    onTap = new PushTapAction(PushTapTarget.fromString(typeFromJSON));
+                }
+
+            } catch (JSONException e){
+                MPLog.d(LOGTAG, "Couldn't parse JSON Object for \'mp_ontap\'");
```

Comment:
if we do end up using this to parse button actions as well, let's make sure we update this comment so it doesn't specify "mp_ontap"